docs/source/topics/storage.rst

```diff
@@ -0,0 +1,89 @@
+.. _topics:storage:
+
+*******
+Storage
+*******
+
+Each AiiDA profile defines a _storage_ which is where all data in the provenance graph is stored.
+Typically, a storage consists of a :ref:`database <topics:database>` and a :ref:`repository <topics:repository>`.
+The provenance graph data itself is mostly persisted to the database, with binary blobs that are attached to nodes, such as files, being stored in the repository.
+
+By default, the storage consists of a PostgreSQL database and a `disk-objectstore container <https://disk-objectstore.readthedocs.io/en/latest/>`_ for the file repository.
+As of AiiDA 2.0, however, this storage can be customized through plugins, meaning other databases and file stores can be used if desired.
+AiiDA ships itself with a number of storage plugins that each have their own strengths and weaknesses.
+This section gives an overview of these storage plugins with suggestions of when to use them.
+
+.. grid:: 1 2 2 2
+   :gutter: 3
+
+   .. grid-item-card:: ``core.psql_dos``
+
+      *Default storage for production projects that require performance.*
+
+      :fa:`database;mr-1` PostgreSQL database
+
+      :fa:`file;mr-1` ``disk-objectstore`` container
+
+      :fa:`plus;mr-1` **Strengths**:
+
+      - Supports all of AiiDA's functionality
+      - Good performance
+      - Automatic database migrations
+
+      :fa:`minus;mr-1` **Weaknesses**:
+
+      - Requires a service running (PostgreSQL)
+
+   .. grid-item-card:: ``core.sqlite_dos``
+
+      *Easy to setup storage for tests, demos and experimenting.*
+
+      :fa:`database;mr-1` SQLite database
+
+      :fa:`file;mr-1` ``disk-objectstore`` container
+
+      :fa:`plus;mr-1` **Strengths**:
+
+      - Easy to setup
+      - Requires no running services
+
+      :fa:`minus;mr-1` **Weaknesses**:
+
+      - Some ``QueryBuilder`` functionality is not supported
+      - Performance of SQLite is inferior to PostgreSQL
```

Comment:
Just for my understanding: what do we mean by "performance" here? It's slower? Or can things go awry?

Reply:
Performance is performance, i.e., it is expected to be slower during operation, especially for larger databases and when it is accessed by multiple processes concurrently, i.e. multiple daemon workers, interactive shells.